### PR TITLE
feat(query): trusted per-key reads for the sum and count+sum aggregate axes

### PR DESCRIPTION
**Pull request by QuantumExplorer**
## The gap

`GroveDb::query_aggregate_count_per_key` existed, but there was no `query_aggregate_sum_per_key` and no `query_aggregate_count_and_sum_per_key`.

This was an implementation gap, not a semantic one — both halves of the work already existed:

- **All three per-key *verifiers* are present and consensus-tested**: `verify_aggregate_count_query_per_key`, `verify_aggregate_sum_query_per_key`, `verify_aggregate_count_and_sum_query_per_key`. A verifier implies a prover, so the carrier descent + aggregate walk was already implemented for all three axes.
- **All three merk primitives are present**, with identical signature shape and the same O(log n) Contained/Disjoint short-circuit: `count_aggregate_on_range`, `sum_aggregate_on_range`, `count_and_sum_aggregate_on_range`.

Only the trusted-read helper was never written for sum and count+sum — the aggregate families landed count-first and each later axis mirrored the *proof* path, because that's what Dash Platform needed verifiable.

## What this does

**1. Extracts a generic carrier walk.** `query_aggregate_carrier_per_key` is one private driver parameterized by which merk aggregate terminates each per-key descent. Leaf-vs-carrier dispatch, the shallow outer-key enumeration via `query_raw` (deliberately not descending into the subquery), `SizedQuery::limit` propagation, the non-tree-match rejection, and the `path / outer_key / subquery_path...` leaf-path assembly are all aggregate-agnostic and now live in exactly one place. The three public entry points pass `Merk::{count,sum,count_and_sum}_aggregate_on_range` as the only axis-specific argument.

The driver is generic over the per-key payload `T` (\`u64\`, \`i64\`, \`(u64, i64)\`) with a single named \`'db\` lifetime, which is what lets the merk methods be passed directly as function items rather than wrapped in closures.

**2. Adds the two missing readers:**

| | returns | leaf host(s) |
|---|---|---|
| `query_aggregate_sum_per_key` | `Vec<(Vec<u8>, i64)>` | `ProvableSumTree`, PCPS |
| `query_aggregate_count_and_sum_per_key` | `Vec<(Vec<u8>, u64, i64)>` | PCPS only |

Both mirror the count entry point's shape, validation (`validate_aggregate_sum_on_range` / `validate_aggregate_count_and_sum_on_range`) and doc conventions — including the \"not independently verifiable, use prove_query + verify_*\" note the existing readers carry. The return types mirror the corresponding per-key verifiers exactly, so the trusted and verified surfaces read the same.

**3. Version gating follows the existing precedent** rather than inventing slots: the count reader already reuses `operations.query.query_aggregate_count_on_range`, so the sum reader reuses `query_aggregate_sum_on_range` and the combined reader `query_aggregate_count_and_sum_on_range`. Purely additive — no existing behavior changes, so nothing needs a V4 gate.

**4. Leaf-shape return convention — kept deliberately, and now documented.** The leaf shape still returns a one-entry vector with an empty stand-in key. That's the convention all three per-key *verifiers* already collapse a leaf proof to, and matching it is what lets a caller swap `query_aggregate_*_per_key` for `prove_query` + `verify_aggregate_*_query_per_key` (or back) and compare element-for-element without branching on shape. A leaf query has no outer key to report, so *some* stand-in is unavoidable; matching the already-shipped proof-side convention is worth more than a prettier one. The doc also notes the disambiguation: outer keys are never empty in a valid carrier, since validation requires every \`subquery_path\` element to be a non-empty key.

## Tests

25 new tests. The load-bearing ones are **differential against the proved path**, since the proof side is already trustworthy: for the same state and the same carrier `PathQuery`, the trusted read must equal `prove_query` + `verify_aggregate_*_query_per_key` element-for-element.

Also covered per axis: leaf and carrier shapes, direction propagation (right-to-left), `limit` capping outer matches *without* capping the inner range, non-tree-match rejection, empty carrier result sets, empty leaf subtrees, and error-surface equality with the corresponding `validate_*` for three malformed-query classes (carrier-with-offset, leaf-with-limit, non-aggregate). The count+sum side additionally asserts the PCPS dual-axis gate fires on the *read* path, not just the proof path.

## Validation

- `cargo clippy --workspace --all-features -- -D warnings` — clean (exit 0)
- `cargo test -p grovedb -p grovedb-query --all-features` — 2614 + 255 passed, 0 failed
- `cargo build --no-default-features --features verify -p grovedb` — clean; the new code is `minimal`-only and no cfg leaked
- `cargo fmt --check` — clean

One note: `cargo clippy --workspace --all-features --all-targets` (not the required invocation) surfaces 5 pre-existing lint failures in `storage/src/rocksdb_storage/tests.rs`, `grovedb-query/src/proofs/mod.rs`, `grovedb-query/src/query_item/mod.rs`, and `grovedb-query/src/proofs/tree_feature_type.rs` — all outside this diff, left alone.

## Out of scope: dispatch wiring

The `run_path_query` `AggregateCarrier` arm returns a typed `NotSupported` for the sum and count+sum kinds and should route to these readers instead — but `run_path_query` lands in #798, which is still open at time of writing. These readers are independent and useful on their own, so the wiring follows as a small follow-up once #798 merges. (Not based on the #798 branch, which is being rebased repeatedly.)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added per-key aggregate sum queries.
  * Added combined per-key count-and-sum queries.
  * Supported leaf and carrier queries, ordering, pagination limits, empty results, and missing keys.
  * Added validation and clear errors for unsupported or invalid query structures.

* **Tests**
  * Added comprehensive coverage for aggregate queries, ordering, limits, validation, and proof-based result comparisons.

* **Documentation**
  * Updated related query documentation.

### CODE REVIEW

File: grovedb/src/tests/aggregate_sum_carrier_query_tests.rs [hunk 1]
```diff
@@ -524,4 +524,423 @@ mod tests {
...
+            vec![TEST_LEAF.to_vec(), b"byBrand".to_vec()],
+            SizedQuery::new(carrier, None, None),
+        );
+        let no_proof = db
+            .grove_db
+            .query_aggregate_sum_per_key(&path_query, None, v)
+            .unwrap()
+            .expect("empty carrier result set must not be an error");
+        assert!(no_proof.is_empty());
+    }
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply grovedb/src/tests/aggregate_sum_carrier_query_tests.rs:746]: This is Claude. Good catch, and correct — the comment claimed the proved path agrees but the test never called `prove_query`.

Fixed in 9af6f8ad by adding the missing differential assertion rather than dropping the claim. An empty carrier result set is exactly the kind of edge case where the trusted read and the proved path could silently diverge (one returning `[]`, the other erroring or emitting a spurious entry), so asserting the agreement is worth more than deleting the sentence. They do in fact agree — `prove_query` succeeds and `verify_aggregate_sum_query_per_key` returns an empty vector too.

Applied the same assertion to the count+sum equivalent (`no_proof_per_key_combined_empty_carrier_result_set`), which had the same gap.

**Review by QuantumExplorer — COMMENTED** [review]
Reviewed
